barretenberg/cpp/src/barretenberg/vm2/common/avm_inputs.test.cpp

```diff
@@ -35,12 +35,81 @@ TEST(AvmInputsTest, FormatTransformations)
     auto flattened = PublicInputs::columns_to_flat(as_cols);
     auto unflattened = PublicInputs::flat_to_columns(flattened);
 
-    ASSERT_THAT(as_cols, SizeIs(1));
-    EXPECT_THAT(as_cols[0], ElementsAre(0));
-    EXPECT_THAT(flattened, ElementsAre(0));
+    ASSERT_THAT(as_cols, SizeIs(AVM_NUM_PUBLIC_INPUT_COLUMNS));
+    for (size_t i = 0; i < AVM_NUM_PUBLIC_INPUT_COLUMNS; ++i) {
+        EXPECT_THAT(as_cols[i], SizeIs(AVM_PUBLIC_INPUTS_COLUMNS_MAX_LENGTH));
+    }
+    EXPECT_THAT(flattened, SizeIs(AVM_PUBLIC_INPUTS_COLUMNS_COMBINED_LENGTH));
 
     EXPECT_EQ(as_cols, unflattened);
 }
 
+TEST(AvmInputsTest, ValuesInColumns)
+{
+    // Create a test PublicInputs with specific values in currently-implemented fields
+    PublicInputs pi;
+
+    // Set global variables
+    pi.globalVariables.blockNumber = 12345;
+
+    // Set start tree snapshots
+    pi.startTreeSnapshots.l1ToL2MessageTree.root = 1000;
+    pi.startTreeSnapshots.l1ToL2MessageTree.nextAvailableLeafIndex = 2000;
+    pi.startTreeSnapshots.noteHashTree.root = 3000;
+    pi.startTreeSnapshots.noteHashTree.nextAvailableLeafIndex = 4000;
+    pi.startTreeSnapshots.nullifierTree.root = 5000;
+    pi.startTreeSnapshots.nullifierTree.nextAvailableLeafIndex = 6000;
+    pi.startTreeSnapshots.publicDataTree.root = 7000;
+    pi.startTreeSnapshots.publicDataTree.nextAvailableLeafIndex = 8000;
+
+    // Set reverted flag
+    pi.reverted = true;
+
+    // Get the columns representation
+    auto columns = pi.to_columns();
+
+    // Convert to flat array for easier testing
+    auto flat = PublicInputs::columns_to_flat(columns);
```

Comment:
best to ASSERT_THAT, the size is correct so that the rest of the test will not blow up when accessing.